### PR TITLE
Add cluster indentifier

### PR DESCRIPTION
**Pull request by tomer-shefler**
…ster cluster

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -206,7 +206,7 @@ CONTROLLER_GEN_OLD ?= $(LOCALBINOLD)/controller-gen
 ENVTEST ?= $(LOCALBIN)/setup-envtest
 
 ## Tool Versions
-KUSTOMIZE_VERSION ?= v3.8.7
+KUSTOMIZE_VERSION ?= v4.5.5
```
Comment (tomer-shefler) [inline Makefile:209]: I ran a diff between the output. The only difference was that description was added

File: main.go [hunk 2]
```diff
@@ -39,20 +41,22 @@ import (
...
 	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
 	ctrl "sigs.k8s.io/controller-runtime"
+
 	"sigs.k8s.io/controller-runtime/pkg/healthz"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
 	operatorcontainerscarbonblackiov1 "github.com/vmware/cbcontainers-operator/api/v1"
 	certificatesUtils "github.com/vmware/cbcontainers-operator/cbcontainers/utils/certificates"
 	"github.com/vmware/cbcontainers-operator/controllers"
-	// +kubebuilder:scaffold:imports
```
Comment (BenRub) [inline main.go:48]: why did you remove this?

File: main.go [hunk 3]
```diff
@@ -91,6 +95,28 @@ func main() {
 		os.Exit(1)
 	}
 
+	setupLog.Info(fmt.Sprintf("Getting Cluster Identifier: %v uid", NamespaceIdentifier))
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		setupLog.Error(err, "unable to get the in cluster rest config")
```
Comment (BenRub) [inline main.go:101]: Why are you not using the existing client like we read Nodes below?

**Review by BenRub — CHANGES_REQUESTED** [review]

File: cbcontainers/communication/gateway/api_gateway.go [hunk 4]
```diff
@@ -90,17 +91,33 @@ func (gateway *ApiGateway) baseRequestWithRetries() *resty.Request {
 }
 
 func (gateway *ApiGateway) getResourcePathWithAccountPath(resourceName string) string {
-	return gateway.baseUrl(fmt.Sprintf("account/%s/%s", gateway.account, resourceName))
+	return gateway.baseUrl(fmt.Sprintf("management/%v", resourceName))
 }
 
-func (gateway *ApiGateway) RegisterCluster() error {
+func (gateway *ApiGateway) SplitToGroupAndMember() (string, string, error) {
```
[the remark this answers is not in the document]
Comment (tomer-shefler) [reply cbcontainers/communication/gateway/api_gateway.go:97]: @yurado wanted to update the API

File: cbcontainers/communication/gateway/api_gateway.go [hunk 5]
```diff
@@ -114,7 +131,7 @@ func (gateway *ApiGateway) RegisterCluster() error {
 }
 
 func (gateway *ApiGateway) GetRegistrySecret() (*models.RegistrySecretValues, error) {
-	url := gateway.getResourcePathWithAccountPath("registrySecret")
+	url := gateway.getResourcePathWithAccountPath("registry_secret")
```
[the remark this answers is not in the document]
Comment (BenRub) [reply cbcontainers/communication/gateway/api_gateway.go:134]: Are you sure? the current operator is working as far as I know

File: cbcontainers/communication/gateway/api_gateway.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (tomer-shefler) [reply cbcontainers/communication/gateway/api_gateway.go:97]: The operator currently used the deprecated API. @yurado Asked me to update the operator to use the new API rather than the deprecate one during this task.

**Review by BenRub — APPROVED** [review]
